src/test/scala/ch/epfl/bluebrain/nexus/admin/routes/ProjectRoutesSpec.scala

```diff
@@ -0,0 +1,269 @@
+      url"http://nexus.example.com/v1/orgs/org".value,
+      orgId,
+      1L,
+      deprecated = false,
+      Set(nxv.Organization.value),
+      instant,
+      caller.subject,
+      instant,
+      caller.subject,
+      Organization("org", "Org description")
+    )
+    val project = Project("label", "org", desc)
+    val resource =
+      ResourceF(iri, projId, 1L, deprecated = false, types, instant, caller.subject, instant, caller.subject, project)
+    val meta = resource.discard
+  }
+
+  "Project routes" should {
+
+    "create a project" in new Context {
+      iamClient.authorizeOn("org" / "label", write) shouldReturn Task.unit
+      iamClient.getCaller(filterGroups = true) shouldReturn Task(caller)
+      projects.create(project) shouldReturn Task(Right(meta))
+
+      Put("/projects/org/label", project.asJson) ~> addCredentials(cred) ~> routes ~> check {
+        status shouldEqual StatusCodes.Created
+        responseAs[Json] shouldEqual meta.asJson
+      }
+    }
+
+    "reject the creation of a project without a label" in new Context {
+      iamClient.authorizeOn(Path("/org").right.value, write) shouldReturn Task.unit
+      iamClient.getCaller(filterGroups = true) shouldReturn Task(caller)
+
+      Put("/projects/org", project.asJson) ~> addCredentials(cred) ~> routes ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        responseAs[Error].code shouldEqual classNameOf[IllegalParameter.type]
+      }
+    }
+
+    "reject the creation of a project with a wrong label" in new Context {
+      iamClient.authorizeOn("org" / "foo", write) shouldReturn Task.unit
+      iamClient.getCaller(filterGroups = true) shouldReturn Task(caller)
+      projects.create(project) shouldReturn Task(Right(meta))
+
+      Put("/projects/org/foo", project.asJson) ~> addCredentials(cred) ~> routes ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        responseAs[Error].code shouldEqual classNameOf[IllegalParameter.type]
+      }
+    }
+
+    "reject the creation of a project which already exists" in new Context {
+      iamClient.authorizeOn("org" / "label", write) shouldReturn Task.unit
+      iamClient.getCaller(filterGroups = true) shouldReturn Task(caller)
+      projects.create(project) shouldReturn Task(Left(ProjectExists))
+
+      Put("/projects/org/label", project.asJson) ~> addCredentials(cred) ~> routes ~> check {
+        status shouldEqual StatusCodes.Conflict
+        responseAs[Error].code shouldEqual classNameOf[ProjectExists.type]
+      }
+    }
+
+    "update a project" in new Context {
+      iamClient.authorizeOn("org" / "label", write) shouldReturn Task.unit
+      iamClient.getCaller(filterGroups = true) shouldReturn Task(caller)
+      projects.update(project, 2L) shouldReturn Task(Right(meta))
+
+      Put("/projects/org/label?rev=2", project.asJson) ~> addCredentials(cred) ~> routes ~> check {
+        status shouldEqual StatusCodes.OK
+        responseAs[Json] shouldEqual meta.asJson
```

Comment:
Same here. It would be better to have on the right hand side directly a static json file instead of `meta.asJson` (which relies on `Encoder`)